src/main/java/com/demo/classroom/Service/AuthService.java

```diff
@@ -124,4 +186,16 @@ private String generateRollNumber() {
 
         return formattedRollNumber;
     }
+
+    public String refreshAccessToken(String refreshToken) {
+    
+        String username = jwtService.extractUsername(refreshToken); 
+        UserDetails userDetails = userService.loadUserByUsername(username);
+
+        if (refreshToken == null || !jwtService.isTokenValid(refreshToken, userDetails)) {
+            return null; 
+        }
```

Comment:
Move the null check for refreshToken to the very start of the method. It's generally better to handle null values as early as possible

Reply:
To verify the the token, we need user details. In the upper part we just extract the userdetails from there.

Reply:
@Boorhan fix this

Reply:
solved